### DISCUSSION TITLE
URL ウィンドウが表示されない

### DISCUSSION THREAD

**Discussion by yarakos95** [increments/qiita-discussions#405] [要望・アイデア・不具合]
PC の Chrome からインストールされた PWA (?) で Qiita を閲覧している場合、左下に表示される URL ウィンドウが表示されません。これは、Chrome から閲覧する場合は表示されます。

ブラウザ|PWA
:--:|:--:
![urlDisplay](https://user-images.githubusercontent.com/75878492/210079971-46e2a385-a53c-4ad4-9ead-b2ba37c73463.png)|![urlNoDisyplay 0](https://user-images.githubusercontent.com/75878492/210079972-352796f4-e05b-4777-a6c1-99094794d56a.png)

この問題は、リンクの表現方法を変えても発生します。
リンクタイトルを指定している場合、ツールチップは表示されています。

<details><summary>コードサンプル</summary>

```markdown
## Link card

https://www.example.com

## Autolinks

<https://www.example.com>

## Link

[Example Domain](https://www.example.com)

### Link with tooltip

[Example Domain](https://www.example.com "This domain is for use in illustrative examples in documents. You may use this domain in literature without prior coordination or asking for permission.")
```

https://qiita.com/Yarakashi_Kikohshi/private/bceed6ba80bc8924986a

</details>

URL が表示されないため、どこに連れ行かれるのか分からず怖いです 😨

**Comment by kato647** [comment]
@Yarakashi-Kikohshi 
調査してみたところ、Chromeのデフォルトの挙動のようでした。
そのため、Qiita側での改善はできないのですがYarakashi-Kikohshiさんの環境を変更いただければ表示が可能そうです。
通常の設定変更ではないため、変更に問題ないか確認していただいてからお試しください。

変更手順
- Chromeのアドレスバーで`chrome://flags/`と入力
- 表示された設定画面で`Desktop PWAs remove status bar`を`Disabled`に変更
- Chromeの再起動

**Reply by yarakos95** [reply]
@kato647 さま

調査のほどありがとうございます。示していただいた変更手順を踏むことで、PWA 内でも URL が表示されるようになりました 🎉

後学のために教えていただきたいのですが、どのようにしてこの問題が Chrome デフォルトの挙動に由来していると分かったのでしょうか。

**Reply by kato647** [reply]
@Yarakashi-Kikohshi 
簡単ではありますが、調査の手順を記載しますね。

- URLが表示されないことが、PWA固有の問題か確認する
  - Chrome以外のブラウザで確認する
  - Safari で確認した時にURLが表示されないため、ブラウザの機能でURLを表示していると判断した
- Safari で表示されないのはなぜか確認する
  - Safari は設定で「ステータスバーの表示・非表示」があることがわかった
- PWAでも設定がないか確認する
  - 「chrome pwa status bar」でGoogle検索して、設定があることを確認できた

**Reply by yarakos95** [reply]
@kato647 さま

ありがとうございます。勉強になります 🖊️

私側の話ですが、私の環境では Chrome と MS Edge を利用しており、両方とも Chromium ベースのブラウザだったため、ブラウザの問題ではないと判断していました 😣

**Comment by yarakos95** [comment]
追記です。

本 Discussion で提起した問題は解決済みでクローズとして大丈夫なのですが、一時、解消されていた問題が再度発生するようになりました。

どうやら最新の Chrome 110 では、提案いただいた Crome flag (`#enable-desktop-pwas-remove-status-bar`) が削除され、PWA 上で URL ウィンドウを表示させる方法は無くなってしまったようです 😢 （PWA はブラウザの性格をなるべく抑えるように設計される方針があるようです。）

> [ChromeOSバージョン109→110にて追加/削除された`chrome://flags`][ref]

[ref]: https://qiita.com/htnk/items/1d2485810ad00fa28397#:~:text=%23enable%2Ddesktop%2Dpwas%2Dremove%2Dstatus%2Dbar

依然として、この問題が Qiita 側ではなくブラウザ側の問題であることは変わりないことが予想されるため、報告を兼ねた追記とさせていただきます。

**Reply by tanaka-asami415** [reply]
@Yarakashi-Kikohshi 
ご報告ありがとうございます！🙇
